apps/mobile/src/features/ConfirmTx/ConfirmTx.container.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react'
+import { ScrollView, View } from 'tamagui'
+import { useScrollableHeader } from '@/src/navigation/useScrollableHeader'
+import { NavBarTitle } from '@/src/components/Title'
+import { TransactionInfo } from './components/TransactionInfo'
+import {
+  MultisigExecutionDetails,
+  useTransactionsGetTransactionByIdV1Query,
+} from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import { useDefinedActiveSafe } from '@/src/store/hooks/activeSafe'
+import { RouteProp, useRoute } from '@react-navigation/native'
+import { ConfirmationView } from './components/ConfirmationView'
+import { LoadingTx } from './components/LoadingTx'
+import { useTxSigner } from './hooks/useTxSigner'
+import { Alert } from '@/src/components/Alert'
+import { ConfirmTxForm } from './components/ConfirmTxForm'
+
+function ConfirmTxContainer() {
+  const activeSafe = useDefinedActiveSafe()
+  const txId = useRoute<RouteProp<{ params: { txId: string } }>>().params.txId
+
+  const { handleScroll } = useScrollableHeader({
+    children: <NavBarTitle paddingRight={5}>Confirm transaction</NavBarTitle>,
+    alwaysVisible: true,
+  })
+
+  const { data, isFetching, isError } = useTransactionsGetTransactionByIdV1Query({
+    chainId: activeSafe.chainId,
+    id: txId,
+  })
+
+  const detailedExecutionInfo = data?.detailedExecutionInfo as MultisigExecutionDetails
```

Comment:
You shouldn't cast like this, use guards instead.

Reply:
@katspaugh you're right, however I found we have a types cohesion problem, where the client gateway is returning a type from a package that is different that we're using in the guards. 

Since it will require us to change some files in the application (because of the types that needs to be changed, I suggest us to handle that in a separated PR, wdyt?

Reply:
Why do you mean? In the web app, there's a `isMultisigExectionDetails` that simply checks the `type` property.
That said, I'm OK with doing it in another PR, this one is too big already.

Reply:
@katspaugh if you take a look in the transaction-guards file, this condition uses some types (`DetailedExecutionInfo, MultisigExecutionDetails`) from the `@safe-global/safe-gateway-typescript-sdk` package, but the type of `data` is from `@safe-global/store/gateway/AUTO_GENERATED/transactions` package, which ended up conflicting with the types in the transaction guard. We need to put the guard to use the types from the  `@safe-global/store/gateway/AUTO_GENERATED/` ones.

Let me know if it is clear, otherwise we can jump in a quick call and I can show to you what I mean.